torch/_dynamo/variables/lists.py

```diff
@@ -308,6 +308,58 @@ def reconstruct(self, codegen):
         ]
         return build_torch_size
 
+    def unpack_var_sequence(self, tx):
+        return [x.add_options(self) for x in self.items]
+
+    def call_method(
+        self,
+        tx,
+        name,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        options = VariableTracker.propagate(self, args, kwargs.values())
+        if name == "__getitem__":
+            assert not kwargs and len(args) == 1
+            if config.dynamic_shapes:
+                out = self.get_item_dyn(tx, args[0])
+            else:
+                out = self.getitem_const(args[0])
```

Comment:
This looks suspicious, we should just guard on index

Reply:
Sorry, can you elaborate?

Reply:
You have some sort of special handling `get_item_dyn` where an indexing expression can be done on a dynamic shape variable, but this is a strange thing to do because we don't support symbolic dim numbers; this would always be a concrete number.

Reply:
This isn't for symbolic dim - this is for getting out symbolic values from a size.

If torch.size() returns [s0, s0, s1], and we slice into it, say [:2], we need a way of producing a SizeVariable of [s0, s1], right? This is that code. 

So even if ndims is concrete, we want to explicitly make the SizeVariable here for this case, or a DynamicShapeVariable for when we only have a single value, right?

Reply:
see https://github.com/pytorch/pytorch/pull/88768#discussion_r1019830454